### PR TITLE
KML Write Function

### PR DESCRIPTION
**Pull request by apurvabanka**
## Summary by Sourcery

Add functionality to write KML and KMZ files with options for pretty printing and XML declaration, and include tests to verify the correct creation of these files.

New Features:
- Introduce a new function to write KML and KMZ files, allowing users to save KML data to specified file paths with options for pretty printing and XML declaration.

Tests:
- Add tests for the new KML write functionality, including tests for writing both KML and KMZ files to ensure files are created correctly.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Introduced a new `write` method in the `KML` class to enable writing KML data to files, supporting both `.kml` and `.kmz` formats.
- The `write` method allows users to specify options for pretty printing and XML declaration.
- Added test cases to verify the functionality of writing KML and KMZ files, ensuring files are created correctly.
- Included a sample KML output file for testing purposes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>kml.py</strong><dd><code>Add KML data writing functionality with file format support</code></dd></summary>
<hr>

fastkml/kml.py

<li>Added a new <code>write</code> method to the <code>KML</code> class for writing KML data to a <br>file.<br> <li> Supports writing to both <code>.kml</code> and <code>.kmz</code> file formats.<br> <li> Includes options for pretty printing and XML declaration.<br>

</details>

  </td>
  <td><a href="https://github.com/cleder/fastkml/pull/378/files#diff-613ee78b8f0776d39e81909e8b514a8053b408d2dc58eb0830deb146a8e12141">+50/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>kml_test.py</strong><dd><code>Add test cases for KML and KMZ file writing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/kml_test.py

<li>Introduced <code>TestWriteKML</code> class for testing KML writing functionality.<br> <li> Added test cases for writing <code>.kml</code> and <code>.kmz</code> files.<br> <li> Verified file creation after writing.<br>

</details>

  </td>
  <td><a href="https://github.com/cleder/fastkml/pull/378/files#diff-601be7e9b008cc98edce432dfce98e46b723b7937152bdeaff5b712d9df56c92">+56/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>output.kml</strong><dd><code>Add sample KML output file for testing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/ogc_conformance/data/kml/output.kml

- Added a sample KML output file for testing purposes.

</details>

  </td>
  <td><a href="https://github.com/cleder/fastkml/pull/378/files#diff-0584b2e827a4eee61828d74ee8cd3be4c980dce7c1430fcf94e1c1a8b7926972">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

## Summary by CodeRabbit

- **New Features**
	- Introduced a method to write KML data to files, supporting both standard KML and compressed KMZ formats.
- **Bug Fixes**
	- Improved error handling during KML serialization.
- **Tests**
	- Added new tests to validate the functionality of writing KML and KMZ files.

### CODE REVIEW

File: fastkml/kml.py [hunk 1]
```diff
@@ -208,6 +210,53 @@ def parse(
...
+                    element,
+                    encoding="UTF-8",
+                    xml_declaration=xml_declaration
+                ).decode(
+                    "UTF-8",
+                )
+
+        if file_path.suffix == ".kmz":
+            with zipfile.ZipFile(file_path, 'w', zipfile.ZIP_DEFLATED) as kmz:
+                kmz.write(file_path, arcname=os.path.basename(file_path))
```
[the remark this answers is not in the document]
Comment (apurvabanka) [reply fastkml/kml.py]: implemented

[the remark this answers is not in the document]
Comment (apurvabanka) [reply fastkml/kml.py]: implemented

File: fastkml/kml.py [hunk 2]
```diff
@@ -208,6 +210,53 @@
...
+        else:
+            with open(file_path, "w") as file:
```
[the remark this answers is not in the document]
Comment (apurvabanka) [reply fastkml/kml.py]: implemented

File: tests/kml_test.py [hunk 3]
```diff
@@ -184,6 +184,62 @@ def test_kml_parse(self) -> None:
...
+                            ns="{http://www.opengis.net/kml/2.2}",
+                        ),
+                    ],
+                    schemata=[],
+                ),
+            ],
+        )
+
+      file_path = KMLFILEDIR / "output.kml"
+      kml.KML.write(doc, file_path=file_path, prettyprint=True, xml_declaration=True)
```
[the remark this answers is not in the document]
Comment (apurvabanka) [reply tests/kml_test.py]: implemented

File: tests/kml_test.py [hunk 4]
```diff
@@ -186,6 +186,61 @@ def test_kml_parse(self) -> None:
...
+                      schemata=[],
+                  ),
+              ],
+          )
+
+        file_path = KMLFILEDIR / "output.kmz"
+
+        doc.write(file_path=file_path, prettyprint=True, xml_declaration=True)
+
+        assert file_path.is_file(), "KMZ file was not created."
```
Comment (cleder) [inline tests/kml_test.py:241]: check that the `kml` is indeed inside the `kmz` file with name of `doc.kml` and the expected contents

Comment (apurvabanka) [reply tests/kml_test.py:248]: Added this check for both the test cases and deleted the output files

**Pull request merged**